src/test/java/it/pagopa/pn/bff/utils/notificationDetail/NotificationDetailUtilityTest.java

```diff
@@ -0,0 +1,297 @@
+                            .recIndex(element.getDetails().getRecIndex())
+                            .documentId(element.getDetails().getGeneratedAarUrl())
+                            .documentType("AAR")
+                            .title("Avviso di avvenuta ricezione - " +
+                                    notificationDTOMultiRecipient.getRecipients().get(element.getDetails().getRecIndex()).getDenomination() +
+                                    "(" + notificationDTOMultiRecipient.getRecipients().get(element.getDetails().getRecIndex()).getTaxId() + ")"
+                            )
+                            .digests(new NotificationAttachmentDigests().sha256(""))
+                            .ref(new NotificationAttachmentBodyRef().key("").versionToken(""))
+                            .contentType("")
+            );
+            index++;
+        }
+    }
+
+    @Test
+    void checkRADDInTimeline() {
+        NotificationDetailTimeline raddFromTimeline = notificationDetailMock.notificationToFERADD().getRadd();
+
+        assert raddFromTimeline.equals(
+                new NotificationDetailTimeline()
+                        .elementId("NOTIFICATION_RADD_RETRIEVED_mock")
+                        .timestamp(OffsetDateTime.parse("2022-06-21T11:44:28Z"))
+                        .legalFactsIds(List.of())
+                        .category(TimelineCategory.NOTIFICATION_RADD_RETRIEVED)
+                        .details(
+                                new NotificationDetailTimelineDetails()
+                                        .recIndex(1)
+                                        .eventTimestamp(OffsetDateTime.parse("2022-06-21T11:44:28Z"))
+                                        .raddType("ALT")
+                                        .raddTransactionId("6")
+                        )
+                        .index(0)
+                        .hidden(true)
+        );
+    }
+
+    @Test
+    void insertCancelledStatusInTimeline() {
+        FullReceivedNotificationV23 cancellationInProgressNotification = new FullReceivedNotificationV23();
+
+        BeanUtils.copyProperties(notificationDetailMock.getOneRecipientNotification(), cancellationInProgressNotification);
+
+        cancellationInProgressNotification.getTimeline().add(
+                notificationDetailMock.getTimelineElem(
+                        TimelineElementCategoryV23.NOTIFICATION_CANCELLATION_REQUEST,
+                        null
+                )
+        );
+
+        BffFullNotificationV1 calculatedParsedNotification = NotificationDetailMapper.modelMapper
+                .mapNotificationDetail(cancellationInProgressNotification);
+
+        NotificationStatusHistory cancellationInProgressStatusHistory =
+                calculatedParsedNotification.getNotificationStatusHistory().stream()
+                        .filter(status -> status.getStatus().equals(NotificationStatus.CANCELLATION_IN_PROGRESS))
+                        .findFirst()
```

Comment:
Same as for others combination of filter and findFirst. Check all the occurrences